modules/abstract-lightning/src/lightning/parseWithdrawPsbt.ts

```diff
@@ -0,0 +1,167 @@
+
+function parsePaymentAddress(script: Buffer, network: utxolib.Network) {
+  for (const type of paymentTypes) {
+    try {
+      let address: string | undefined;
+      switch (type) {
+        case 'p2pkh':
+          address = utxolib.payments.p2pkh({ output: script, network }).address;
+          break;
+        case 'p2wpkh':
+          address = utxolib.payments.p2wpkh({ output: script, network }).address;
+          break;
+        case 'p2wsh':
+          address = utxolib.payments.p2wsh({ output: script, network }).address;
+          break;
+        case 'p2sh':
+          address = utxolib.payments.p2sh({ output: script, network }).address;
+          break;
+        case 'p2tr':
+          address = utxolib.payments.p2tr({ output: script, network }).address;
+          break;
+        default:
+          continue;
+      }
+      if (address) {
+        return address;
+      }
+    } catch (e) {
+      // Do nothing, just try the next type
+    }
+  }
+  return '';
+}
+
+function parsePsbtOutputs(psbt: Psbt, network: utxolib.Network): WithdrawBaseOutputUTXO<bigint>[] {
+  const parsedOutputs: WithdrawBaseOutputUTXO<bigint>[] = [];
+  let bip32Derivation: Bip32Derivation | undefined;
+
+  for (let i = 0; i < psbt.data.outputs.length; i++) {
+    const output = psbt.data.outputs[i];
+    const txOutput = psbt.txOutputs[i];
+
+    let address = '';
+    const value = txOutput.value;
+    let isChange = false;
+
+    if (output.bip32Derivation && output.bip32Derivation.length > 0) {
+      isChange = true;
+      bip32Derivation = output.bip32Derivation[0];
+    }
+
+    if (txOutput.script) {
+      address = parsePaymentAddress(txOutput.script, network);
+    }
+    const valueBigInt = BigInt(value);
+
+    parsedOutputs.push({
+      address,
+      value: valueBigInt,
+      change: isChange,
+      bip32Derivation,
+    });
+  }
+
+  return parsedOutputs;
+}
+
+function verifyChangeAddress(
+  output: WithdrawBaseOutputUTXO<bigint>,
+  accounts: WatchOnlyAccount[],
+  network: utxolib.Network
+): void {
+  if (!output.bip32Derivation || !output.bip32Derivation.path) {
+    throw new Error(`bip32Derivation path not found for change address`);
+  }
+  const derivationPath = output.bip32Derivation.path;
+
+  // Extract the purpose from the derivation path (e.g., m/84' -> 84)
+  const pathSegments = derivationPath.split('/');
+  const purpose = Number(pathSegments[1].replace("'", ''));
+
+  // Find the corresponding account using the purpose
+  const account = accounts.find((acc) => acc.purpose === purpose);
+  if (!account) {
+    throw new Error(`Account not found for purpose: ${purpose}`);
+  }
+
+  // Create a BIP32 node from the xpub
+  const xpubNode = utxolib.bip32.fromBase58(account.xpub, network);
+
+  // Extract the change and address_index from the path (e.g., m/84'/0'/0'/1/0 -> 1 and 0)
+  const change = Number(pathSegments[pathSegments.length - 2]);
+  const addressIndex = Number(pathSegments[pathSegments.length - 1]);
```

Comment:
we're reading a lot of structure into these derivation paths, it probably makes sense to have a `parseDerivationPath(path: string): { purpose: number; isChange: boolean; addressIndex: number } ` with tests